tests/lib/rules/lines-around-comment.js

```diff
@@ -89,6 +89,201 @@ eslintTester.addRuleTest("lib/rules/lines-around-comment", {
+            options: [{
+                beforeLineComment: true,
+                allowBlockStart: true
+            }]
+        },
+        {
+            code: "if(true){\n\n// line at block start\nvar g = 1;\n}",
+            options: [{
+                beforeLineComment: true,
+                allowBlockStart: true
+            }]
+        },
+        {
+            code: "function foo(){   \n/* block comment at block start */\nvar g = 1;\n}",
+            options: [{
+                allowBlockStart: true
+            }]
+        },
+        {
+            code: "function foo(){/* block comment at block start */\nvar g = 1;\n}",
+            options: [{
+                allowBlockStart: true
+            }]
+        },
+        {
+            code: "var foo = function(){\n/* block comment at block start */\nvar g = 1;\n}",
+            options: [{
+                allowBlockStart: true
+            }]
+        },
+        {
+            code: "if(true){\n/* block comment at block start */\nvar g = 1;\n}",
+            options: [{
+                allowBlockStart: true
+            }]
+        },
+        {
+            code: "if(true){\n\n/* block comment at block start */\nvar g = 1;\n}",
+            options: [{
+                allowBlockStart: true
+            }]
+        },
+        {
+            code: "while(true){\n\n/* \nblock comment at block start\n */\nvar g = 1;\n}",
+            options: [{
+                allowBlockStart: true
+            }]
+        },
+
+        // check for block end comments
+        {
+            code: "function foo(){\nvar g = 91;\n// line at block end\n}",
+            options: [{
+                afterLineComment: true,
+                allowBlockEnd: true
+            }]
+        },
+        {
+            code: "function foo(){\nvar g = 61;\n\n\n// line at block end\n}",
+            options: [{
+                afterLineComment: true,
+                allowBlockEnd: true
+            }]
+        },
+        {
+            code: "var foo = function(){\nvar g = 1;\n\n\n// line at block end\n}",
+            options: [{
+                afterLineComment: true,
+                allowBlockEnd: true
+            }]
+        },
+        {
+            code: "if(true){\nvar g = 1;\n// line at block end\n}",
+            options: [{
+                afterLineComment: true,
+                allowBlockEnd: true
+            }]
+        },
+        {
+            code: "if(true){\nvar g = 1;\n\n// line at block end\n}",
+            options: [{
+                afterLineComment: true,
+                allowBlockEnd: true
+            }]
+        },
+        {
+            code: "while(true){\n// line at block start and end\n}",
+            options: [{
+                afterLineComment: true,
+                allowBlockEnd: true
+            }]
+        },
+        {
+            code: "while(true){\n// line at block start and end\n}",
```

Comment:
Given that `allowBlockStart` and `allowBlockEnd` are not tied together in any fashion, this test is equivalent to the same test without `allowBlockStart` enabled and therefore  should be changed in favor of one that tests with all 4 options combined, i.e. :

```js
{
    code: "while(true){\n// line at block start and end\n}",
    options: [{
        afterLineComment: true,
        beforeLineComment: true,
        allowBlockStart: true,
        allowBlockEnd: true
    }]
},
```